src/vs/workbench/services/workingCopy/common/workingCopyBackupTracker.ts

```diff
@@ -159,13 +159,16 @@ export abstract class WorkingCopyBackupTracker extends Disposable {
 			return;
 		}
 
-		// Schedule backup if dirty
-		if (workingCopy.isDirty()) {
+		// Schedule backup for unsaved content
+		if (workingCopy.isModified()) {
 			// this listener will make sure that the backup is
 			// pushed out for as long as the user is still changing
 			// the content of the working copy.
 			this.scheduleBackup(workingCopy);
 		}
+		else {
```

Comment:
This sounds like a dangerous change to me, why?

Reply:
yes I could see some unintended consequences from this. The reason was that we only discard backups in the onDidChangeDirty, which scratchpads will never trigger, meaning that we would never discard scratchpad backups. Is that OK?